comit/src/orderpool.rs

```diff
@@ -90,6 +91,23 @@ impl OrderPool {
             .flat_map(|(maker, orders)| iter::from_fn(move || Some(maker)).zip(orders.values()))
     }
 
+    pub fn safe_expiries(&self) -> impl Iterator<Item = (&PeerId, &BtcDaiOrder)> {
```

Comment:
Hmm, can we do better with the naming here? The function reads like an accessor that returns you expiries but it actually returns orders where the expiries match!

I see two options:

1. Completely imply this check on our view on the orderbook. Going that way I would just add this check to the "all" iterator and hence we can just avoid the naming discussion.
2. Provide a dedicated accessor like here but change the name so that it is clear that we are returning orders here.

The downside of approach (1) is that we are putting this policy deep into the orderpool and it will get in the way once we want to give the user the option to also see orders that don't match their expiries.
We are not there yet but I guess we can take it as a hint that we might want to keep this out of the orderpool?

Re option (2): Maybe a name like "executable_orders" conveys better what we are returning here? A Rust-doc comment might also help :)

Reply:
I'm in favour of option 2. I don't quite understand what you mean by approach 1 "putting this policy deep into the orderpool". Wouldn't approach 1 be moving the policy (return only executable orders) out of the orderpool?

Reply:
If you provide an accessor on orderpool, the orderpool has knowledge about this policy of what an executable order is although that policy is actually something that only the caller is concerned about. Note that while the orderpool cares about the swap protocol for matching, it will happily publish and receive orders of with all kinds of swap protocols. We choose the offsets in the HTTP controller and hand the complete order over to the pool for publishing. One could argue that the choice of which orders are returned to the user should happen on the same level and hence not be part of the orderpool :)

Reply:
yep understood. Got confused by this line "The downside of approach (1) is that we are putting this policy deep into the orderpool". I think you meant approach(2). Ill move the filtering into the request handler

Reply:
Yeah, filtering in the HTTP controller makes sense! It is slightly less efficient if we cannot compose the iterators in one go but currently, the orderpool is so small that it doesn't really matter :)